src/main/java/com/google/devtools/build/lib/runtime/StarlarkOptionsParser.java

```diff
@@ -228,7 +228,7 @@ private Target loadBuildSetting(
     ImmutableList.Builder<String> keep = ImmutableList.builder();
     ImmutableList.Builder<String> remove = ImmutableList.builder();
     for (String name : list) {
-      ((name.startsWith("--//") || name.startsWith("--no//")) ? remove : keep).add(name);
+      (name.matches("--(no)?(@.*)?//.*") ? remove : keep).add(name);
```

Comment:
Regex-based parsing is going to fall into difficulties. Can this just strip off the -- and the no (if present), and call [LabelValidator.validateAbsoluteLabel](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/cmdline/LabelValidator.java;drc=61d663a503710c8cb0d03d18f2eb776981b49e17;l=227), and check for exceptions being thrown?

Reply:
So much better. Done.